crates/nargo_cli/src/cli/codegen_verifier_cmd.rs

```diff
@@ -32,34 +37,51 @@ pub(crate) fn run<B: Backend>(
     args: CodegenVerifierCommand,
     config: NargoConfig,
 ) -> Result<(), CliError<B>> {
-    // TODO(#1201): Should this be a utility function?
-    let circuit_build_path = args
-        .circuit_name
-        .map(|circuit_name| config.program_dir.join(TARGET_DIR).join(circuit_name));
+    let workspace = resolve_workspace(&config.program_dir)?;
 
-    let common_reference_string = read_cached_common_reference_string();
+    for package in &workspace {
+        let circuit_build_path = workspace.package_build_path(package);
+
+        let smart_contract_string = smart_contract_for_package(
+            backend,
+            package,
+            circuit_build_path,
+            &args.compile_options,
+        )?;
+
+        let contract_dir = workspace.contracts_directory_path();
+        create_named_dir(&contract_dir, "contract");
+        let contract_path = contract_dir.join("plonk_vk").with_extension("sol");
```

Comment:
Does this need the package name now?

Reply:
Don't think so since you are selecting a package?

Reply:
`contracts_dir` is rooted in the workspace, not the package. We could change that to make a separate contracts directory rooted in each package, but I think that is wrong. @TomAFrench thoughts?

Reply:
I temporarily decided to add another directory into the path, so `<root>/crates/a` would produce `<root>/contract/a/plonk_vk.sol`

We can change if folks thinks that's not ideal.